nemo/collections/tts/modules/fastpitch.py

```diff
@@ -230,19 +236,27 @@ def forward(
 
         # Input FFT
         enc_out, enc_mask = self.encoder(input=text, conditioning=spk_emb)
-
-        log_durs_predicted = self.duration_predictor(enc_out, enc_mask)
+        if self.speaker_emb_condition_prosody:
+            log_durs_predicted = self.duration_predictor(enc_out + spk_emb, enc_mask)
+        else:
+            log_durs_predicted = self.duration_predictor(enc_out, enc_mask) 
```

Comment:
We could reduce code duplication by rewriting the if conditions like:

```
if self.speaker_emb_condition_prosody:
  prosody_input = enc_out + spk_emb
else:
  prosody_input = enc_out
log_durs_predicted = self.duration_predictor(prosody_input, enc_mask)
pitch_predicted = self.pitch_predictor(prosody_input, enc_mask)
```

Reply:
agreed! tts collections have too many duplicated snippets. If we are not ready to refactor them now, it'd be better to avoid duplicate codes sooner or later.

Reply:
Cool, I will add this change.